File: java/src/main/java/com/gaston/macbook/simplemvp/presentation/show_live_iss/view/ISSLive.java
package com.gaston.macbook.simplemvp.presentation.show_live_iss.view;


import android.os.Bundle;
import android.view.Window;
import android.view.WindowManager;
import android.widget.Toast;
import com.gaston.macbook.simplemvp.BuildConfig;
import com.gaston.macbook.simplemvp.R;
import com.google.android.youtube.player.YouTubeBaseActivity;
import com.google.android.youtube.player.YouTubeInitializationResult;
import com.google.android.youtube.player.YouTubePlayer;
import com.google.android.youtube.player.YouTubePlayerView;
import butterknife.BindView;
import butterknife.ButterKnife;

public class ISSLive extends YouTubeBaseActivity {

    @BindView(R.id.isslive_youtube_player)
    YouTubePlayerView youtubePlayerView;

    private static final String VIDEO_ID = "dVTzaXmYm_s";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        requestWindowFeature(Window.FEATURE_NO_TITLE);
        getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN);
        setContentView(R.layout.activity_isslive);
        ButterKnife.bind(this);
        initVideo();

    }

    private void initVideo() {
        youtubePlayerView.initialize(BuildConfig.YoutubeApiKey, new YouTubePlayer.OnInitializedListener() {
            @Override
            public void onInitializationSuccess(YouTubePlayer.Provider provider, YouTubePlayer youTubePlayer, boolean b) {
                if (!b){
                    youTubePlayer.loadVideo(VIDEO_ID);
                    youTubePlayer.setPlayerStyle(YouTubePlayer.PlayerStyle.DEFAULT);
                }
            }

            @Override
            public void onInitializationFailure(YouTubePlayer.Provider provider, YouTubeInitializationResult youTubeInitializationResult) {
                Toast.makeText(ISSLive.this, getString(R.string.youtube_video_load_failed), Toast.LENGTH_SHORT).show();
            }
        });
    }
}

File: kotlin/src/main/java/com/gaston/macbook/kotlinsimplemvp/presentation/show_live_iss/view/ISSLive.kt
package com.gaston.macbook.kotlinsimplemvp.presentation.show_live_iss.view

import android.os.Bundle
import android.view.Window
import android.view.WindowManager
import android.widget.Toast
import com.gaston.macbook.kotlinsimplemvp.BuildConfig
import com.gaston.macbook.kotlinsimplemvp.R
import com.google.android.youtube.player.YouTubeBaseActivity
import com.google.android.youtube.player.YouTubeInitializationResult
import com.google.android.youtube.player.YouTubePlayer
import kotlinx.android.synthetic.main.activity_isslive.*

class ISSLive : YouTubeBaseActivity() {

    companion object {
        const val VIDEO_ID = "dVTzaXmYm_s"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        requestWindowFeature(Window.FEATURE_NO_TITLE)
        window.setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN)
        setContentView(R.layout.activity_isslive)
        initVideo()
    }

    private fun initVideo() {
        isslive_youtube_player.initialize(BuildConfig.YoutubeApiKey, object : YouTubePlayer.OnInitializedListener {
            override fun onInitializationSuccess(provider: YouTubePlayer.Provider, youTubePlayer: YouTubePlayer, b: Boolean) {
                if (!b) {
                    youTubePlayer.loadVideo(ISSLive.VIDEO_ID)
                    youTubePlayer.setPlayerStyle(YouTubePlayer.PlayerStyle.DEFAULT)
                }
            }

            override fun onInitializationFailure(provider: YouTubePlayer.Provider, youTubeInitializationResult: YouTubeInitializationResult) {
                Toast.makeText(this@ISSLive, getString(R.string.youtube_video_load_failed), Toast.LENGTH_SHORT).show()
            }
        })
    }
}
